src/streamlink_cli/main.py

```diff
@@ -123,7 +127,17 @@ def create_http_server(host=None, port=0):
 
     return http
 
-
+def create_title(plugin=None):
```

Comment:
Looks like there is a bug in the Python 2 part, `args.encode('utf8')`. 

I think you could simplify this method by encoded (or not) at the end - didn't test it though. 
```python
def create_title(plugin=None):
	if args.title and plugin:
        title = args.title.format(title=plugin._get_title(),
                                  author=plugin._get_author(),
                                  category=plugin._get_category(),
                                  game=plugin._get_category())
    else:
        title = args.url

	if is_py2:
        return title.encode("utf8")
	else:
        return title
```

Reply:
Also, what should happen in the case where `_get_author` is not implemented?

Reply:
It's supposed to fallback to these values:

     DEFAULT_STREAM_METADATA = {
         "title"    : u"Untitled",
         "author"   : u"Unknown Author",
         "category" : u"No Category",
         "game"     : u"No Game/Category"
     }

Reply:
I can't see where that happens...

Reply:
Maybe not implemented yet...

Reply:
I changed plugin.get_author, plugin.get_title, and plugin.get_category to no longer `raise NotImplementedError`. Now it'll return the values in `DEFAULT_STREAM_METADATA`.